### PR TITLE
change info log to debug

### LINKED ISSUE #1541: estimateGas shows out of gas error
#### System information

Geth
Version: 1.10.2-stable
Git Commit: e5b930e8e9a6ff043b3e71a4e3b852280fe413f0
Quorum Version: 22.7.0
Architecture: amd64
Network Id: 1337
Go Version: go1.16.15
Operating System: linux
GOPATH=
GOROOT=go

#### Expected behaviour

No error is displayed when running estimateGas.

#### Actual behaviour

An error is displayed when running estimateGas.
In fact, estimateGas returned the correct gas estimate and we were able to execute the transaction within that estimate.
So this is a display issue only.

#### Steps to reproduce the behaviour

1. Install GoQuorum with Quickstart.
> npx quorum-dev-quickstart
スクリーンショット 2022-10-18 144913
2. Start the installed GoQuorum.
> cd ./quorum-test-network/
> sudo ./run.sh
3. Open docker terminal for rpcnode.
4. Perform geth attach on rpcnode's terminal.
> geth attach --datadir /data
5. Run estimateGas for the contract creation transaction.
> eth.estimateGas({"from":"0xDEADBEEFCAFE0000000000000000000000000000","data":"0x608060405234801561001057600080fd5b50336000806101000a81548173ffffffffffffffffffffffffffffffffffffffff021916908373ffffffffffffffffffffffffffffffffffffffff16021790555060008054906101000a900473ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16600073ffffffffffffffffffffffffffffffffffffffff167f342827c97908e5e2f71151c08502a66d44b6f758e3ac2f1de95f02eb95f0a73560405160405180910390a3610356806100db6000396000f3fe608060405234801561001057600080fd5b50600436106100365760003560e01c8063893d20e81461003b578063a6f9dae114610059575b600080fd5b610043610075565b604051610050919061025d565b60405180910390f35b610073600480360381019061006e91906101fe565b61009e565b005b60008060009054906101000a900473ffffffffffffffffffffffffffffffffffffffff16905090565b60008054906101000a900473ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff163373ffffffffffffffffffffffffffffffffffffffff161461012c576040517f08c379a000000000000000000000000000000000000000000000000000000000815260040161012390610278565b60405180910390fd5b8073ffffffffffffffffffffffffffffffffffffffff1660008054906101000a900473ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff167f342827c97908e5e2f71151c08502a66d44b6f758e3ac2f1de95f02eb95f0a73560405160405180910390a3806000806101000a81548173ffffffffffffffffffffffffffffffffffffffff021916908373ffffffffffffffffffffffffffffffffffffffff16021790555050565b6000813590506101f881610309565b92915050565b600060208284031215610214576102136102db565b5b6000610222848285016101e9565b91505092915050565b610234816102a9565b82525050565b6000610247601383610298565b9150610252826102e0565b602082019050919050565b6000602082019050610272600083018461022b565b92915050565b600060208201905081810360008301526102918161023a565b9050919050565b600082825260208201905092915050565b60006102b4826102bb565b9050919050565b600073ffffffffffffffffffffffffffffffffffffffff82169050919050565b600080fd5b7f43616c6c6572206973206e6f74206f776e657200000000000000000000000000600082015250565b610312816102a9565b811461031d57600080fd5b5056fea26469706673582212208f65d9c920a5e7c951f66594688d8b1bb886420d647e05d3e5c9ec7eeab019cd64736f6c63430008070033","value":"0x0"})
スクリーンショット 2022-10-18 145657
6. "contract creation code storage out of gas" appears in rpcnode stdout.
> INFO [10-18|05:56:40.268] VM returned with error                   err="contract creation code storage out of gas"
スクリーンショット 2022-10-18 145733

### PR DESCRIPTION
**Pull request by baptiste-b-pegasys**
avoid displaying all vm errors returned by estimate gas and all other cases by using debug level
fixes #1541

### CODE REVIEW

File: core/state_transition.go [hunk 1]
```diff
@@ -285,6 +287,7 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 			publicState.SetNonce(sender.Address(), publicState.GetNonce(sender.Address())+1)
 		}
 		if err != nil {
+			log.Warn("execution result", "err", err)
```
Comment (baptiste-b-pegasys) [inline core/state_transition.go]:
```suggestion
```

File: core/state_transition.go [hunk 2]
```diff
@@ -312,9 +315,11 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	// Check clauses 4-5, subtract intrinsic gas if everything is correct
 	gas, err := IntrinsicGas(st.data, st.msg.AccessList(), contractCreation, homestead, istanbul)
 	if err != nil {
+		log.Warn("intrinsicgas", "gas", gas, "err", err)
```
Comment (baptiste-b-pegasys) [inline core/state_transition.go]:
```suggestion
```

File: core/state_transition.go [hunk 3]
```diff
@@ -312,9 +315,11 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
...
 		return nil, err
 	}
 	if st.gas < gas {
+		log.Warn("intrinsicgas", "st.gas", st.gas, "gas", gas)
```
Comment (baptiste-b-pegasys) [inline core/state_transition.go]:
```suggestion
```

File: core/state_transition.go [hunk 4]
```diff
@@ -340,6 +345,7 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	)
 	if contractCreation {
 		ret, _, leftoverGas, vmerr = evm.Create(sender, data, st.gas, st.value)
+		log.Warn("contract creation", "ret", ret, "err", err, "leftovergas", leftoverGas)
```
Comment (baptiste-b-pegasys) [inline core/state_transition.go]:
```suggestion
```

File: core/state_transition.go [hunk 5]
```diff
@@ -415,11 +421,13 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	}
 	// End Quorum
 
-	return &ExecutionResult{
+	er := &ExecutionResult{
```
Comment (baptiste-b-pegasys) [inline core/state_transition.go]:
```suggestion
	return &ExecutionResult{
```

File: core/state_transition.go [hunk 6]
```diff
@@ -415,11 +421,13 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
...
 		UsedGas:    st.gasUsed(),
 		Err:        vmerr,
 		ReturnData: ret,
-	}, nil
+	}
+	log.Warn("execution result", "er", er)
+	return er, nil
```
Comment (baptiste-b-pegasys) [inline core/state_transition.go]:
```suggestion
	}, nil
```

**Review by baptiste-b-pegasys — COMMENTED** [review]
remove unwanted changes

**Review by antonydenyer — APPROVED** [review]

**Pull request merged**